modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessWrappedException.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.painless;
+
+/**
+ * Checked exceptions are wrapped in {@link ClassValue}#getFromHashMap in Error
+ * which leads to unexpected behavior in Painless. This class is used as a
+ * workaround for that exception wrapping.
+ */
+public class PainlessWrappedException extends Error {
```

Comment:
Is there a reason why this extends error and not RuntimeException? (I suppose RuntimeException would not work with ClassValue.getFromHashMap, but checking)

Reply:
Painless allows users to catch `Exception`, so in cases where we do not want users to be able to recover from an exception we use `Error` instead. I realize this is a slight change in behavior from before, but I consider the change to fix a bug since catching an unresolvable conversion in this case will just lead to the script doing this every execution.